components/camel-spring-rabbitmq/src/main/java/org/apache/camel/component/springrabbit/DefaultMessagePropertiesConverter.java

```diff
@@ -39,12 +41,68 @@ public DefaultMessagePropertiesConverter(CamelContext camelContext, HeaderFilter
     @Override
     public MessageProperties toMessageProperties(Exchange exchange) {
         MessageProperties answer = new MessageProperties();
-        String contentType = ExchangeHelper.getContentType(exchange);
+        Message message = exchange.getMessage();
+
+        Object deliveryMode = message.getHeader(SpringRabbitMQConstants.DELIVERY_MODE);
+        if (deliveryMode != null) {
+            answer.setDeliveryMode(deliveryMode instanceof MessageDeliveryMode ? (MessageDeliveryMode) deliveryMode
+                    : MessageDeliveryMode.fromInt(Integer.parseInt(deliveryMode.toString())));
```

Comment:
Let's see what others say but I'm not really a fan of calling `toString` without even knowing/checking the type

Reply:
right this is ambiguous especially since this is an enum... I simplifed that and added the enums for the doc.